In [1]:
print('hello')

hello


In [2]:
print('new')

new


here you can write notes

In [3]:
print('new')

new


In [4]:
2+1

3

In [5]:
2-1

1

In [6]:
2*2

4

In [7]:
3/2

1.5

Modulo or "Mod" Operator

In [8]:
7/4

1.75

In [9]:
7%4

3

In [11]:
50%5

0

In [13]:
20 %2

0

In [14]:
2 ** 3

8

In [16]:
2 +10 *10 + 3

105

In [1]:
10*10

100

LECTURE STARTS FOR NUMBER & STRINGS

In [11]:
a = 5

In [12]:
a

5

In [13]:
a = 10

In [14]:
a

10

In [15]:
a + a

20

In [19]:
a = a + a

In [20]:
a

80

In [21]:
type(a)

int

In [22]:
a = 30.1

In [23]:
type(a)

float

In [24]:
my_income = 100
tax_rate = 0.1
my_taxes = my_income * tax_rate

In [25]:
my_taxes

10.0

NEXT CHAPTER IS ABOUT STRINGS

# PYTHON IN BUILD FUNCTION DEMO

In [ ]:
mylist = [1,2,3]

# RANGE FUNCTIO
range(x,y)
range(10) start frm 0-10
range(3,10) strart frm 3-10

In [3]:
for num in range(2):
    print(num)for i in range(len(list)):

0
1


In [4]:
for num in range(3,10):
    print(num)

3
4
5
6
7
8
9


# USING EVEN NUMBER WITH RANGE BELOW IS THE EXAMPLE

In [5]:
for num in range(0,10,2):
    print(num)

0
2
4
6
8


# USING ODD NUMBER WITH RANGE BELOW IS THE EXAMPLE

In [6]:
for num in range(0,10,3):
    print(num)

0
3
6
9


# DIRECT ON COMMAND RUN USE LIST RANGE BELOW IS THE EXAMPLE

In [7]:
list(range(0,10,2))

[0, 2, 4, 6, 8]

# ENUMERATE FUNCTION IN PYTHON

In [13]:
index_count = 0
for letter in 'abcd':
    print('At index {} the letter is {}'.format(index_count,letter))
    index_count+=1

At index 0 the letter is a
At index 1 the letter is b
At index 2 the letter is c
At index 3 the letter is d


In [2]:
index_count = 0
word = 'abcde'
for letter in word:
    print(word[index_count])
    index_count +=1

a
b
c
d
e


In [5]:
word = 'abcde'
for item in enumerate(word):
    print(item)

(0, 'a')
(1, 'b')
(2, 'c')
(3, 'd')
(4, 'e')


# SO THIS IS BASICALLY DOING THAT INDEX COUNT FOR US AUTOMATICALLY IN THE FORM OF TUPLES.

In [8]:
word = 'abcde'
for index,letter in enumerate(word):
    print(index,letter)
    #print(letter)

0 a
1 b
2 c
3 d
4 e


# ABOVE IS ANOTHER WAY TO SOLVE THE QUERY

In [ ]:
word = 'abcde'
for index,letter in enumerate(word):
    print(index)
    print(letter)
    print('\n')

# ABOVE IS ANOTHER WAY TO SOLVE THE QUERY

# NEW FUCNTION ZIPS BELOW IS THE EXAMPLE

In [10]:
mylist = [1,2,3]
mylist1 = ['a','b','c']
for item in zip(mylist,mylist1):
    print(item)

(1, 'a')
(2, 'b')
(3, 'c')


# ZIPS FUNCTION COMBINE 2 VALUE AND DISPLAY IN TUPLES.
SO PREVIOUS IN THAT FOR LOOP LECTURE I MENTIONED THAT TUPLE AND PACKING IS GOING TO BE REALLY USEFUL

In [11]:
mylist = [1,2,3]
mylist1 = ['a','b','c']
mylist2 = [100,200,300]
for item in zip(mylist,mylist1,mylist2):
    print(item)

(1, 'a', 100)
(2, 'b', 200)
(3, 'c', 300)


In [13]:
mylist = [1,2,3]
mylist1 = ['a','b','c']
mylist2 = [100,200,300]
for a,b,c in zip(mylist,mylist1,mylist2):
    print(c)

100
200
300


##### " in "  is operator and HOW TO CHECK IF THE VALUE IS IN THE LIST.

In [14]:
'x' in [1,2,3]

False

In [15]:
'x' in ['X','Y','Z']

False

In [16]:
'x' in ['x','y','z']

True

# MATH LIBRARY

In [18]:
mylist = [10,20,30,40,100]
min(mylist)

10

In [19]:
mylist = [10,20,30,40,100]
max(mylist)

100

# HOW TO IMPORT THE LIBRARY IN PYTHON AS UNDER.

In [1]:
from random import shuffle
mylist = [1,2,3,4,5,6,7,8,9,10]
print(mylist)
shuffle(mylist)
mylist

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


[10, 1, 7, 4, 3, 5, 9, 8, 2, 6]

#### shuffle not returning anything
so if i asked for this list as equal to shuffle that's not going to work because it's not returning anything
# example below

In [2]:
random_list = shuffle(mylist)

In [3]:
random_list

# random_list do not retrun any thing because just use type command to see the result below

In [4]:
type(random_list)

NoneType

# above example return nothing 
# so it is an in-place function meaning it operate in place on that list

In [5]:
mylist

[6, 9, 7, 5, 10, 4, 2, 8, 3, 1]

# THERE IS ALSO NICE FUNCTION FOR GRABBING A RANDOM INTEGER

In [6]:
from random import randint
randint(0,100)

12

In [7]:
randint(0,100)

94

In [8]:
mynum = randint(0,10)
mynum

6

# HOW TO GET INPUT FROM THE USER BELOW IS THE EXAMPLE
# FUNCTIN IS INPUT

In [9]:
input('Enter a number here :')

Enter a number here :123


'123'

# HOW TO STORE VALUE FROM USER AND PRINT THEM BELOW IS THE EXAMPLE.

In [11]:
result = input('What is your name ?')
result

What is your name ?Jose


'Jose'

In [12]:
type(result)

str

In [14]:
result = input('What is your Number')
result

What is your Number30


'30'

In [15]:
float(result)

30.0

In [16]:
int(result)

30

# SO KEEP IN MIND IMPA ALWAYS ACCEPT THINGS AS A STRING BECAUSE LATER ON YOU MAY WANT TO BE USEING
# SO YOU WILL LIKELY HAVE TO CAST IT AND YOU CAN DO THIS ALL IN ONE LINE BY SAYING WRAP THIS IN I.A.

In [17]:
result = int(input('What is your Number'))
result

What is your Number30


30

In [18]:
type(result)

int

STRINGS

In [1]:
'hello'

'hello'

In [2]:
"world"

'world'

In [3]:
'This is also a string'

'This is also a string'

In [6]:
'I'm going on run'

SyntaxError: invalid syntax (<ipython-input-6-8d2173b8d3c8>, line 1)

ABOVE CODE GIVE ERROR BECAUSE ''END ' SO BELOW IS THE RIGHT SYNTAX

In [5]:
"I'm going to run"

"I'm going to run"

2 line to be display we have use the syntax print() live example

In [7]:
"hello world one"
"hello world two"

'hello world two'

In [8]:
print('hello world one')
print('hello world two')

hello world one
hello world two


HOW TO PRINT IN NEW LINE USING PRINT SYNTAX US \n (CASE SENSATIVE)

In [10]:
print('hello \n world')

hello 
 world


in above example we want to maintain margin then use below syntax.

In [11]:
print('hello \nworld')

hello 
world


in above example we have maintain margin then use above syntax.

how to use the tab syntax using print statement

In [13]:
print('hello \tworld')

hello 	world


CHECK THE LENGHT OF THE STRING MENTION BELOW

In [15]:
len('hello')

5

In [16]:
len('I am')

4

i am having 4 character (single space white character is count

In [2]:
mystring = "Hello World"

In [18]:
mystring

'Hello World'

SINGLE CHARACTER TO PRINT SYNTAX IS [WHICH NUMBER TO SELECTED] NUMBER ALWAY STARTS WIHT 0

In [3]:
mystring[0]

'H'

In [4]:
mystring[8]

'r'

In [5]:
mystring[9]

'l'

In [7]:
mystring

'Hello World'

In [8]:
mystring[-2]

'l'

NEW TOPIC SLICING

In [9]:
mystring = 'abcdefghijk'

In [10]:
mystring

'abcdefghijk'

In [11]:
mystring[2]

'c'

In [12]:
mystring[2:]

'cdefghijk'

In [15]:
mystring[:3]

'abc'

In [16]:
mystring[3:6]

'def'

In [18]:
mystring[1:3]

'bc'

In [19]:
mystring[::]

'abcdefghijk'

THIS IS VALID SYNTAX TO DISPLAY THE ENTIRE mystring[::]

In [20]:
mystring[::2]

'acegik'

THIS WILL DISPLAY THE mystring[::2] FRM 0-2 / 2-4 AND SO ON (JUMPING SIZE OF 2

In [21]:
mystring[2:7:2]

'ceg'

STRING FRM 2 THEN 7 AND AFTER 7 JUMP TO 2

REVERSE STEP STRING JUMPING

In [22]:
mystring[::-1]

'kjihgfedcba'

ENTIRE STRING WILL GET REVERSE

In [23]:
mystring = "Hello World"

In [28]:
"Hello World"[8]

'r'

In [1]:
"tinker"[1:4]

'ink'

Format with the .format() method
A good way to format object into your string for print statement is with the string.format()
method. The syntax is:
'String here {} then also {}'.format('something1','something2')

In [3]:
print('This is the string {}'.format('INSERTED'))

This is the string INSERTED


In [4]:
print('The {} {} {}'.format('Fox','Brown','Quick'))

The Fox Brown Quick


In [5]:
print('The {2} {1} {0}'.format('Fox','Brown','Quick'))

The Quick Brown Fox


In [6]:
print('The {0} {0} {0}'.format('Fox','Brown','Quick'))

The Fox Fox Fox


HOW TO ASSING VARIABLE IN PRINT STATEMENT BELOW IS THE EXAMPLE

In [7]:
print('The {q} {b} {f}'.format(f='fox',b='Brown',q='Quick'))

The Quick Brown fox


In [8]:
print('The {f} {f} {f}'.format(f='fox',b='Brown',q='Quick'))

The fox fox fox


Float Formatting follows "{value:width.precision f}"

In [10]:
result = 100/777

In [11]:
result

0.1287001287001287

In [13]:
print("The result was {r}".format(r=result))

The result was 0.1287001287001287


In [14]:
print("The result was {r:1.3f}".format(r=result))

The result was 0.129


IN PRINT DECIMAL HOW MUCH WHITE SPACE U NEED TO DISPLAY AS MENTION BELOW

In [15]:
print("The result was {r:10.3f}".format(r=result))

The result was      0.129


HOW MUCH DECIMAL TO BE DISPLAY MENTIN BELOW

In [16]:
print("The result was {r:1.5f}".format(r=result))

The result was 0.12870


In [17]:
result = 104.12345

In [19]:
print("The result was {r:1.2f}".format(r=result))

The result was 104.12


NEW STRING METHOD INTRODUCE IN PYTHON 3.6 MENTION BELOW

In [20]:
name = 'Jose'

In [21]:
print('hello the name is {}'.format(name))

hello the name is Jose


In [22]:
print(f'hello the name is {name}')

hello the name is Jose


In [23]:
name = 'Sam'
age = 3

In [25]:
print(f'{name} is {age}')

Sam is 3


In [27]:
print(f'{name} is {age} years old')

Sam is 3 years old


In [29]:
'I like {}'.format('apples')

'I like apples'

In [37]:
'I like apples ' + 'Python rules!'

'I like apples Python rules!'

# Strings

Strings are used in Python to record text information, such as names. Strings in Python are actually a *sequence*, which basically means Python keeps track of every element in the string as a sequence. For example, Python understands the string "hello' to be a sequence of letters in a specific order. This means we will be able to use indexing to grab particular letters (like the first letter, or the last letter).

This idea of a sequence is an important one in Python and we will touch upon it later on in the future.

In this lecture we'll learn about the following:

    1.) Creating Strings
    2.) Printing Strings
    3.) String Indexing and Slicing
    4.) String Properties
    5.) String Methods
    6.) Print Formatting

## Creating a String
To create a string in Python you need to use either single quotes or double quotes. For example:

In [1]:
# Single word
'hello'

'hello'

In [2]:
# Entire phrase 
'This is also a string'

'This is also a string'

In [3]:
# We can also use double quote
"String built with double quotes"

'String built with double quotes'

In [4]:
# Be careful with quotes!
' I'm using single quotes, but this will create an error'

SyntaxError: invalid syntax (<ipython-input-4-da9a34b3dc31>, line 2)

The reason for the error above is because the single quote in <code>I'm</code> stopped the string. You can use combinations of double and single quotes to get the complete statement.

In [5]:
"Now I'm ready to use the single quotes inside a string!"

"Now I'm ready to use the single quotes inside a string!"

Now let's learn about printing strings!

## Printing a String

Using Jupyter notebook with just a string in a cell will automatically output strings, but the correct way to display strings in your output is by using a print function.

In [6]:
# We can simply declare a string
'Hello World'

'Hello World'

In [7]:
# Note that we can't output multiple strings this way
'Hello World 1'
'Hello World 2'

'Hello World 2'

We can use a print statement to print a string.

In [8]:
print('Hello World 1')
print('Hello World 2')
print('Use \n to print a new line')
print('\n')
print('See what I mean?')

Hello World 1
Hello World 2
Use 
 to print a new line


See what I mean?


## String Basics

We can also use a function called len() to check the length of a string!

In [9]:
len('Hello World')

11

Python's built-in len() function counts all of the characters in the string, including spaces and punctuation.

## String Indexing
We know strings are a sequence, which means Python can use indexes to call parts of the sequence. Let's learn how this works.

In Python, we use brackets <code>[]</code> after an object to call its index. We should also note that indexing starts at 0 for Python. Let's create a new object called <code>s</code> and then walk through a few examples of indexing.

In [10]:
# Assign s as a string
s = 'Hello World'

In [11]:
#Check
s

'Hello World'

In [12]:
# Print the object
print(s) 

Hello World


Let's start indexing!

In [13]:
# Show first element (in this case a letter)
s[0]

'H'

In [14]:
s[1]

'e'

In [15]:
s[2]

'l'

We can use a <code>:</code> to perform *slicing* which grabs everything up to a designated point. For example:

In [16]:
# Grab everything past the first term all the way to the length of s which is len(s)
s[1:]

'ello World'

In [17]:
# Note that there is no change to the original s
s

'Hello World'

In [18]:
# Grab everything UP TO the 3rd index
s[:3]

'Hel'

Note the above slicing. Here we're telling Python to grab everything from 0 up to 3. It doesn't include the 3rd index. You'll notice this a lot in Python, where statements and are usually in the context of "up to, but not including".

In [19]:
#Everything
s[:]

'Hello World'

We can also use negative indexing to go backwards.

In [20]:
# Last letter (one index behind 0 so it loops back around)
s[-1]

'd'

In [21]:
# Grab everything but the last letter
s[:-1]

'Hello Worl'

We can also use index and slice notation to grab elements of a sequence by a specified step size (the default is 1). For instance we can use two colons in a row and then a number specifying the frequency to grab elements. For example:

In [22]:
# Grab everything, but go in steps size of 1
s[::1]

'Hello World'

In [23]:
# Grab everything, but go in step sizes of 2
s[::2]

'HloWrd'

In [24]:
# We can use this to print a string backwards
s[::-1]

'dlroW olleH'

## String Properties
It's important to note that strings have an important property known as *immutability*. This means that once a string is created, the elements within it can not be changed or replaced. For example:

In [25]:
s

'Hello World'

In [26]:
# Let's try to change the first letter to 'x'
s[0] = 'x'

TypeError: 'str' object does not support item assignment

Notice how the error tells us directly what we can't do, change the item assignment!

Something we *can* do is concatenate strings!

In [27]:
s

'Hello World'

In [28]:
# Concatenate strings!
s + ' concatenate me!'

'Hello World concatenate me!'

In [29]:
# We can reassign s completely though!
s = s + ' concatenate me!'

In [30]:
print(s)

Hello World concatenate me!


In [31]:
s

'Hello World concatenate me!'

We can use the multiplication symbol to create repetition!

In [32]:
letter = 'z'

In [33]:
letter*10

'zzzzzzzzzz'

## Basic Built-in String methods

Objects in Python usually have built-in methods. These methods are functions inside the object (we will learn about these in much more depth later) that can perform actions or commands on the object itself.

We call methods with a period and then the method name. Methods are in the form:

object.method(parameters)

Where parameters are extra arguments we can pass into the method. Don't worry if the details don't make 100% sense right now. Later on we will be creating our own objects and functions!

Here are some examples of built-in methods in strings:

In [34]:
s

'Hello World concatenate me!'

In [35]:
# Upper Case a string
s.upper()

'HELLO WORLD CONCATENATE ME!'

In [36]:
# Lower case
s.lower()

'hello world concatenate me!'

In [37]:
# Split a string by blank space (this is the default)
s.split()

['Hello', 'World', 'concatenate', 'me!']

In [38]:
# Split by a specific element (doesn't include the element that was split on)
s.split('W')

['Hello ', 'orld concatenate me!']

There are many more methods than the ones covered here. Visit the Advanced String section to find out more!

## Print Formatting

We can use the .format() method to add formatted objects to printed string statements. 

The easiest way to show this is through an example:

In [39]:
'Insert another string with curly brackets: {}'.format('The inserted string')

'Insert another string with curly brackets: The inserted string'

We will revisit this string formatting topic in later sections when we are building our projects!

## Next up: Lists!

HOW TO CHANGE AND JOIN THE STRING STEP MENTION BELOW

In [2]:
name = "Sam"

In [4]:
#name[0] = "P"

In [11]:
last_letters = name[1:]

In [12]:
last_letters

'am'

In [13]:
'P' + last_letters

'Pam'

In [14]:
x = "Hello World"

In [16]:
x + ' it is beautify outsite!'

'Hello World it is beautify outsite!'

In [17]:
x = x + ' it is beautify outsite!'

In [18]:
x

'Hello World it is beautify outsite!'

In [19]:
letter = 'z'

In [20]:
letter * 10

'zzzzzzzzzz'

In [21]:
2 + 3

5

In [22]:
'2' + '3'

'23'

ABOVE EXAMPLE HOW TO ADD NUMBER DATA TYPE AND STRING DATA TYPE

NEW TOPI ABOUT FUNCTION (STRING IN BUILT FUNCTION) x. press TAB AND WILL SEE THE FUCNTION AVAILABLE IN EDITOR

x = "Hello World"

In [25]:
x.upper()

'HELLO WORLD IT IS BEAUTIFY OUTSITE!'

In [26]:
x.lower()

'hello world it is beautify outsite!'

In [27]:
x.split()

['Hello', 'World', 'it', 'is', 'beautify', 'outsite!']

In [28]:
x = 'hi this is a string'

In [29]:
x.split()

['hi', 'this', 'is', 'a', 'string']

In [30]:
x.split('i')

['h', ' th', 's ', 's a str', 'ng']

    CHAPTER BOOLEANS

BOOLEAN ARE OPERATORS THAT ALLOW YOU TO CONVEY TRUE OR FALSE STATEMENTS.

THESE ARE VERY IMPORTANT LATER ON WE DEAL WITH CONTROL FLOW AND LOGIC
BOOLEAN VARIABLE = True or False

In [2]:
True

True

type will show the boolean key word as mention below

In [3]:
type(True)

bool

In [4]:
type(False)

bool

In [5]:
1 > 2

False

EQUAL SIGN == BELOW IS THE EXAMPLE

In [6]:
1 == 1

True

In [ ]:
None is operator have NULL OR NOTHING vale as mention below

In [7]:
B = None

In [8]:
B

In [10]:
type({1,2,3,4})

set

In [11]:
set([1,1,2,3])

{1, 2, 3}

In [1]:
2 == 2

True

In [2]:
2 ==1

False

In [3]:
'hello' == 'bye'

False

In [4]:
'Bye' == 'bye'

False

In [5]:
'2' == 2

False

In [6]:
2.00 == 2

True

In [8]:
3 != 3

False

In [9]:
2 >= 2

True

In [10]:
4 <= 1

False

MULTIPLE OPERATOR AND OR NOT

In [11]:
1 < 2

True

In [12]:
2 < 3

True

In [14]:
1 < 2 & 2 < 3

True

In [3]:
1 < 2 and 2 > 3

False

In [4]:
1 < 2 and 2 < 3

True

In [6]:
'h' == 'h' and 2 == 2

True

In [7]:
('h' == 'h') and (2 == 2)

True

OR OPERATION ANY ONE CONDITION HAS TO BE TRUE

In [8]:
1 == 1 or 2 == 2

True

NOT OPERATOR ASK TO RETURN THE OPPOSITE WHAT U DID

In [10]:
not 1 == 1

False

In [11]:
400 > 5000

False

In [12]:
not 400 > 5000

True

DICTIONAY SYNTAX IS AS FOLLOW :-
    {'key1':'value1',key2':'value2'}
    curley bracket open 'field1':'value1','field2':'value2' curley bracket closed

    DIFFERENCE BETWEEN LIST AND DICTIONARY
A) DICTIONARY : OBJECT RETRIVE BY KEY NAME. UNORDER AND CAN BE SORTED
B) LIST       : OBJECT RETRIVE BY LOCATION. ORDER SEQUENCE CAN BE INDEXED OR SLICED.

EXAMPLE BELOW

In [1]:
my_dict = {'key1':'value1','key2':'value2'}

In [2]:
my_dict

{'key1': 'value1', 'key2': 'value2'}

In [4]:
my_dict['key1']

'value1'

In [6]:
price_lookup = {'apple':2.99,'oranges':1.99,'milk':5.80}

In [8]:
price_lookup['apple']

2.99

    MOST IMPORTANT EXAMPE

In [9]:
d = {'k1':123,'k2':[0,1,2],'k3':{'insidekey':100}}

In [11]:
d['k2']

[0, 1, 2]

In [12]:
d['k3']

{'insidekey': 100}

In [13]:
d['k3']['insidekey']

100

HOW TO DISPLAY THE VALUE K2 AND [] VALUE AS MENTION BELOW

In [14]:
d['k2']

[0, 1, 2]

In [15]:
d['k2'][2]

2

In [16]:
d={'key1':['a','b','c']}

In [17]:
d

{'key1': ['a', 'b', 'c']}

    DICTIONAY VALUE TO BE CONVERT IN LIST SEE THE STEPS

In [18]:
my_list = d['key1']

In [19]:
my_list

['a', 'b', 'c']

In [23]:
letter = my_list[2]

In [24]:
letter.upper()

'C'

SECOND WAY TO CHANGE THE LOWER CASE TO UPPER CASE AS UNDER

In [26]:
d['key1'][2].upper()

'C'

In [28]:
d = {'k1':100,'k2':200}

In [29]:
d['k3']=300

In [30]:
d

{'k1': 100, 'k2': 200, 'k3': 300}

OVERWRITE THE VALUE IN DICTIONARY

In [31]:
d['k1']='NEW VALUE'

In [32]:
d

{'k1': 'NEW VALUE', 'k2': 200, 'k3': 300}

In [1]:
d = {'k1':100,'k2':200,'k3':300}

In [2]:
d

{'k1': 100, 'k2': 200, 'k3': 300}

In [3]:
d.keys()

dict_keys(['k1', 'k2', 'k3'])

ABOVE FUNCTION d.key() SHOW THE DICTIONARY KEYS

In [4]:
d.values()

dict_values([100, 200, 300])

ABOVE FUNCTION d.values() SHOW THE DICTIONARY VALUES

In [5]:
d.items()

dict_items([('k1', 100), ('k2', 200), ('k3', 300)])

ABOVE FUNCTION d.items() SHOW THE DICTIONARY KEYS & VALUE

In [8]:
{'Monday':19,'Tuesday':20}

{'Monday': 19, 'Tuesday': 20}

In [9]:
d={'k1':[1,2,3]}

In [10]:
d['k1'][1]

2

SETS

1. SETS ARE UNORDERD COLLECTION OF UNIQUE ELEMENTS.

MEANING THE CAN ONLY BE ONE REPRESENTATIVE OF THE SAME OBJECT.

LET'S SEE SOME EXAMPLES.!

In [5]:
myset = set()

ABOVE VARIBALE SET TO DISPLAY WILL BE EMPTY

In [3]:
set()

set()

HOW TO ADD VALUE IN THE SETS EXAMPLE GIVEN BELOW

In [6]:
myset.add(1)

In [9]:
myset

{1, 2}

now notice it looks like it's a dictionary because it is curley braces but it's not because it doesn't have a key value pairs
it's is onordered collection of the items you want
so it's an other object here.

In [8]:
myset.add(2)

now try to adding 2 a second time and see what happens

In [10]:
myset.add(2)

In [11]:
myset

{1, 2}

NOW TAKE A LOOK I HAVE ADD 2 AS SECOND TIME BUT IT DOES NOT ADD
I CALL MY SET IS STILL JUST SHOW 1 AND 2 IT DOSEN'T SHOW 1 2 2 AND THIS IS BASICALLY WHAT IS IT

    UNIQUE VALUE IN THE SET
BELOW IS THE EXAMPLES

In [12]:
mylist =[1,1,1,1,1,2,2,2,3,3,3,3]

In [13]:
set(mylist)

{1, 2, 3}

PLEASE NOTE THAT ALL DUPLICATE VALUE HAVE BEEN IGNORED.

In [14]:
set('Mississippi')

{'M', 'i', 'p', 's'}

NEW CHAPTER " TUPLES "
TUPLES ARE VERY SIMILAR TO LISTS. HOWEVER THEY HAVE ONE KEY DIFFERENCE - IMMUTABILITY

ONCE AN ELEMENT IS INSIDE A TUPLE, IT CAN NOT BE REASSIGNED.

TUPLES USED PARENTHESIS : (1,2,3)

In [2]:
t = (1,2,3)

In [3]:
my_list = [1,2,3]

THE BELOW EXAMPLE WILL RETURN TYPE

In [4]:
type(t)

tuple

ANOTHER VARIBALE IS my_list SEE BELOW WHAT IT RETURNS

In [6]:
type(my_list)

list

CHECK LENGTH OF TUPLES AND LIST MENTION BELOW

In [7]:
len(t)

3

In [9]:
t

(1, 2, 3)

In [10]:
t = ('one',2)

In [11]:
t[0]

'one'

In [12]:
t[-2]

'one'

In [13]:
t = ('a','a','b')

AND LET IMAGINE WE WANT TO COUNT HOW MANY TIME THE LETTER H SHOWS UP IN THE TUPLES

In [14]:
t.count('a')

2

USING INDEX WITH TUPLES

In [15]:
t.index('a')

0

In [16]:
t.index('b')

2

WE CAN REAASING THE VALUE IN LIST BELOW IS THE EXAMPLE

In [17]:
my_list[0]='NEW'

In [18]:
my_list

['NEW', 2, 3]

FOR TUPLE REAASIGN THE VALUE BELOW IS THE ERROR

In [19]:
t[0]='NEW'

TypeError: 'tuple' object does not support item assignment

In [20]:
(1,2,[1,2])

(1, 2, [1, 2])

In [21]:
type((1,2,[1,2]))

tuple

In [23]:
type([1,2])

list

In [1]:
my_list = [1,2,3]

In [2]:
my_list = ['STRING',100,23.2]

In [3]:
len(my_list)

3

In [4]:
mylist=['one','two','three']

In [5]:
mylist[0]

'one'

In [6]:
mylist[1:]

['two', 'three']

In [7]:
another_list = ['four','five']

HOW TO COMBILE LIST

In [8]:
mylist + another_list

['one', 'two', 'three', 'four', 'five']

COMBINE LIST THE WE HAVE TO USE ANOTHER VARIABLE AND COMINE MYLIST + ANOTHER_LIST

In [9]:
new_list = mylist + another_list

In [10]:
new_list

['one', 'two', 'three', 'four', 'five']

HOW TO CHANGE THE LIST VALUE (EXAMPLE = 'one' to 'ONE ALL CAPS'

In [11]:
new_list[0] = 'ONE ALL CAPS'

In [12]:
new_list

['ONE ALL CAPS', 'two', 'three', 'four', 'five']

HOW THE ADD,APPEND,DELETE ETC IN LIST USING VARIABLE NANE THEN . AND PRESS TAB

In [13]:
new_list.append('six')

In [14]:
new_list

['ONE ALL CAPS', 'two', 'three', 'four', 'five', 'six']

In [15]:
new_list.append('seven')

In [16]:
new_list

['ONE ALL CAPS', 'two', 'three', 'four', 'five', 'six', 'seven']

DELETE FRM LIST USING FUNCTION POP

In [17]:
new_list.pop

<function list.pop(index=-1, /)>

In [18]:
new_list.pop()

'seven'

USING POP FUNCTION LAST ITEM WILL BE REMOVED SEE ABOVE EXAMPLE AND BELOW IS RESULT WHAT HAPPEN

In [19]:
new_list

['ONE ALL CAPS', 'two', 'three', 'four', 'five', 'six']

NOTE LAST ITEM LIST IS POP
USING POP FUNCTION (IN OTHER WORDS DELETE) SAVE IN NEW VARIABLE THE CAN BE DISPLAY

In [20]:
popped_item = new_list.pop()

In [21]:
popped_item

'six'

In [22]:
new_list

['ONE ALL CAPS', 'two', 'three', 'four', 'five']

NEW_LIST VARIBALE HAS BEEN DISPLAY AND SIX HAS BEEN REMOVED.
IN new_list last item 'six' has been remove and save in varibale popped_item

POP FUNCTION IS HERE <function list.pop(index=-1, /)>
DEFAULT POP LIST ARE AT THE (-1) LAST ITEM BUT IN LIST ITEM FRM 0 - N ANY ITEM CAN BE POP BELOW IS EXAMPLE

In [23]:
new_list.pop(0)

'ONE ALL CAPS'

In [24]:
new_list

['two', 'three', 'four', 'five']

In [1]:
new_list = ['a','e','x','b','c']
num_list=[4,1,8,3]

SORTING THE LIST VARIBLE

In [2]:
new_list

['a', 'e', 'x', 'b', 'c']

In [3]:
new_list.sort()

In [4]:
new_list

['a', 'b', 'c', 'e', 'x']

In [7]:
my_sorted_list = new_list.sort()

ASSING NEW VARIABLE IN USING SORT FUNCTION WILL NOT DISPLAY ANY THING OR U TYPE my_sorted_list = nontype()
IT MEANS NON TYPE OBJECT
None type mean no value (it is used as placeholder)

In [9]:
new_list.sort()
my_sorted_list = new_list

    ABOVE IS HOW THE SORTED LIST STORE IN THE VARIBALE
    AND BELOW IS THE RESULT

In [10]:
my_sorted_list

['a', 'b', 'c', 'e', 'x']

In [11]:
num_list

[4, 1, 8, 3]

In [12]:
num_list.sort()

    IN LIST HOW NUMBER IS SORTED

In [13]:
num_list

[1, 3, 4, 8]

    NEXT IN LIST IS REVERSE METHOD

In [14]:
num_list.reverse()

In [15]:
num_list

[8, 4, 3, 1]

In [18]:
[1,'two',3.14159]

[1, 'two', 3.14159]

In [19]:
lst=['a','b','c']

#### LIST COMPREHENSION ARE A UNIQUE WAY OF QUICKLY CREATING LIST WITH PYTHON.
#### IF YOU FIND YOU FIND YOURSELF USING A FOR LOOP ALONG WITH .append() TO CREATE LIST, List
#### COMPREHENSION ARE A GOOD ALTERNATIVE.

In [3]:
mystring = 'hello'
mylist = []
for letter in mystring:
    mylist.append(letter)
mylist

['h', 'e', 'l', 'l', 'o']

# SAME CAN BE DONE IN SINGLE LINE BELOW IS THE EXAMPLE

In [5]:
mylist = [letter for letter in mystring]
mylist

['h', 'e', 'l', 'l', 'o']

In [6]:
mylist =[x for x in 'word']
mylist

['w', 'o', 'r', 'd']

In [7]:
mylist = [x for x in range(0,11)]
mylist

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [8]:
mylist = [x**2 for x in range(0,11)]
mylist

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81, 100]

In [12]:
mylist = [x for x in range(0,11)if x%2==0]
mylist

[0, 2, 4, 6, 8, 10]

# another way to do it below

In [13]:
mylist = [x for x in range(0,11,2)if x%2==0]
mylist

[0, 2, 4, 6, 8, 10]

In [16]:
mylist = [x**2 for x in range(0,11,2)if x%2==0]
mylist
type(mylist)

list

In [15]:
celcius = [0,10,20,34.5]
faherenheit = [((9/5)*temp+32) for temp in celcius]
faherenheit

[32.0, 50.0, 68.0, 94.1]

In [17]:
faherenheit = []
for temp in celcius:
    faherenheit.append(((9/5)*temp+32))
faherenheit

[32.0, 50.0, 68.0, 94.1]

#### REALLY UGLY AND IT'S HARD TO READ WHEN YOU COME BACK TO THEM A MONTH LATER

#### SO KEEP THAT IN MIND

#### TUTOR IS GOING TO SHOW YOU RIGHT NOW HOW TO IT BUT I AM MYSELF TYPICALLY DON'T DO THIS BECAUSE IT 'S REALLY HARD TO READ

###### ONCE YOU COME BACK TO IT LET ME SHOW YOU HOW TO USE AND IF AND ELSE STATEMENT INSIDE OF LIST COMPRESSION

In [5]:
result = [x if x%2==0 else 'ODD' for x in range(0,11)]
result

[0, 'ODD', 2, 'ODD', 4, 'ODD', 6, 'ODD', 8, 'ODD', 10]

##### so this kind of weired because now the order has been totally reveresed

## NESTED FOR LOOPS

In [1]:
mylist = []
for x in [2,4,6]:
    for y in [100,200,300]:
        mylist.append(x*y)
print(mylist)

[200, 400, 600, 400, 800, 1200, 600, 1200, 1800]


In [2]:
mylist = []
for x in [2,4,6]:
    for y in [1,10,1000]:
        mylist.append(x*y)
print(mylist)

[2, 20, 2000, 4, 40, 4000, 6, 60, 6000]


In [4]:
mylist = [x*y for x in [2,4,6]for y in [1,10,1000]]
mylist

[2, 20, 2000, 4, 40, 4000, 6, 60, 6000]

Use for, .split(), and if to create a Statement that will print out words that start with 's':

In [ ]:
st = 'Print only the words that start with s in this sentence'
#st = st.split()
    print(x)

In [12]:
y = [x for x in range(0,10)if x%2==0]
y

[0, 2, 4, 6, 8]

In [17]:
y = [x for x in range(0,50) if x%3==0]
y

[0, 3, 6, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42, 45, 48]

### Go through the string below and if the length of a word is even print "even!"

In [52]:
st = 'Print every word in this sentence that has an even number of letters'
for x in st.split():
    if len(x)%2 ==0:
        print(x,"even")

word even
in even
this even
sentence even
that even
an even
even even
number even
of even


### Write a program that prints the integers from 1 to 100. But for multiples of three print "Fizz" instead of the number, and for the multiples of five print "Buzz". For numbers which are multiples of both three and five print "FizzBuzz".

In [77]:
x=0
for x in range(0,100):
    if x % 3==0 and x % 5 !=0:
        print(x,"Fizz")
    elif x % 5==0 and x % 3 !=0:
        print(x,"Buzz")
    elif x % 3==0 and x % 5==0:
        print(x,"FizzBuzz")

0 FizzBuzz
3 Fizz
5 Buzz
6 Fizz
9 Fizz
10 Buzz
12 Fizz
15 FizzBuzz
18 Fizz
20 Buzz
21 Fizz
24 Fizz
25 Buzz
27 Fizz
30 FizzBuzz
33 Fizz
35 Buzz
36 Fizz
39 Fizz
40 Buzz
42 Fizz
45 FizzBuzz
48 Fizz
50 Buzz
51 Fizz
54 Fizz
55 Buzz
57 Fizz
60 FizzBuzz
63 Fizz
65 Buzz
66 Fizz
69 Fizz
70 Buzz
72 Fizz
75 FizzBuzz
78 Fizz
80 Buzz
81 Fizz
84 Fizz
85 Buzz
87 Fizz
90 FizzBuzz
93 Fizz
95 Buzz
96 Fizz
99 Fizz


### Use List Comprehension to create a list of the first letters of every word in the string below:

In [1]:
st = 'Create a list of the first letters of every word in this string'
[word[0] for word in st.split()]

['C', 'a', 'l', 'o', 't', 'f', 'l', 'o', 'e', 'w', 'i', 't', 's']

SYNTAX OF IF STATEMENT
IF 
ELIF
ELSE

if some_condition :
    # execute some code
else:
    # execute some code

------------------------------- MULTIPLE IF STATEMENT STATEMENT
if some_condition :
    # execute some code
elif:
    # execute some code
else:
    # execute some code
    

In [2]:
if True:
    print('ITS TRUE!')

ITS TRUE!


In [4]:
if 3>2:
    print('ITS TRUE!')

ITS TRUE!


In [6]:
hungry = True
if hungry:
    print('FEED ME!')

FEED ME!


In [7]:
hungry = False
if hungry:
    print('FEED ME!')
else:
    print('I AM NOT HUNGRY')

I AM NOT HUNGRY


In [8]:
loc = 'Bank'
if loc == 'Auto Shop':
    print('Car are cool!')
else:
    print('I do not know much')

I do not know much


USING ELIF OPERATION IN IF - ELIF - ELSE 

In [10]:
loc = 'Bank'
if loc == 'Auto shop':
    print('Car are cool!')
elif loc == 'Bank':
    print('Money is cool!')
elif loc == 'Store' :
    print('Welcome to store')
else:
    print('I do not know much')

Money is cool!


In [12]:
loc = 'Auto shop'
if loc == 'Auto shop':
    print('Car are cool!')
elif loc == 'Bank':
    print('Money is cool!')
elif loc == 'Store' :
    print('Welcome to store')
else:
    print('I do not know much')

Car are cool!


In [13]:
loc = 'Store'
if loc == 'Auto shop':
    print('Car are cool!')
elif loc == 'Bank':
    print('Money is cool!')
elif loc == 'Store' :
    print('Welcome to store')
else:
    print('I do not know much')

Welcome to store


In [14]:
loc = 'Game'
if loc == 'Auto shop':
    print('Car are cool!')
elif loc == 'Bank':
    print('Money is cool!')
elif loc == 'Store' :
    print('Welcome to store')
else:
    print('I do not know much')

I do not know much


In [15]:
name = 'Sammy'
if name == 'Frankie!':
    print("Hello Frankie!")
elif name == 'Sammy':
    print("Hello Sammy!")
else:
    print("What is your name")

Hello Sammy!


In [17]:
name = 'Frankie!'
if name == 'Frankie!':
    print("Hello Frankie!")
elif name == 'Sammy':
    print("Hello Sammy!")
else:
    print("What is your name")

Hello Frankie!


In [18]:
name = 'Jose'
if name == 'Frankie!':
    print("Hello Frankie!")
elif name == 'Sammy':
    print("Hello Sammy!")
else:
    print("What is your name")

What is your name


# FIRST ASSIGNMENT TEST ON LINE

In [9]:
st = 'Print only the words that start with s in this sentence'
for word in st.split():
    if word[0] == 's':
        print(word)

start
s
sentence


###### Use range() to print all the even from 0 to 10 

In [11]:
for num in range(0,11,2):
#    if num % 2 == 0:
        print(num)

0
2
4
6
8
10


###### Use a List Comprehension to create a list of all numbers between 1 to 50 that are divisible by 3

In [15]:
[x for x in range(1,51)if x %3==0]

[3, 6, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42, 45, 48]

###### Go throught the string below and if the length of a word is even print even

In [18]:
st = 'Print every word in this sentence that has even number of letters'
for word in st.split():
    if len(word)% 2 == 0:
        print(word,'is even')

word is even
in is even
this is even
sentence is even
that is even
even is even
number is even
of is even


#### FIZZ

In [ ]:
for num in range(1,101):
    if num % 3 == 0 and num % 5 == 0:
        print(num,'FizzBuzz')
    elif num % 3 ==0:
        print(num,'Fizz')
    elif num % 5 ==0:
        print(num,'Buzz')
    else :
        print(num)

# Statements Assessment Solutions

**Use <code>for</code>, .split(), and <code>if</code> to create a Statement that will print out words that start with 's':**

In [1]:
st = 'Print only the words that start with s in this sentence'

In [2]:
for word in st.split():
    if word[0] == 's':
        print(word)

start
s
sentence


______
**Use range() to print all the even numbers from 0 to 10.**

In [3]:
list(range(0,11,2))

[0, 2, 4, 6, 8, 10]

___
**Use List comprehension to create a list of all numbers between 1 and 50 that are divisible by 3.**

In [4]:
[x for x in range(1,51) if x%3 == 0]

[3, 6, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42, 45, 48]

_____
**Go through the string below and if the length of a word is even print "even!"**

In [5]:
st = 'Print every word in this sentence that has an even number of letters'

In [6]:
for word in st.split():
    if len(word)%2 == 0:
        print(word+" <-- has an even length!")

word <-- has an even length!
in <-- has an even length!
this <-- has an even length!
sentence <-- has an even length!
that <-- has an even length!
an <-- has an even length!
even <-- has an even length!
number <-- has an even length!
of <-- has an even length!


____
**Write a program that prints the integers from 1 to 100. But for multiples of three print "Fizz" instead of the number, and for the multiples of five print "Buzz". For numbers which are multiples of both three and five print "FizzBuzz".**

In [ ]:
for num in range(1,101):
    if num % 3 == 0 and num % 5 == 0:
        print("FizzBuzz")
    elif num % 3 == 0:
        print("Fizz")
    elif num % 5 == 0:
        print("Buzz")
    else:
        print(num)

____
**Use a List Comprehension to create a list of the first letters of every word in the string below:**

In [7]:
st = 'Create a list of the first letters of every word in this string'

In [8]:
[word[0] for word in st.split()]

['C', 'a', 'l', 'o', 't', 'f', 'l', 'o', 'e', 'w', 'i', 't', 's']

### Great Job!

# Statements Assessment Test
Let's test your knowledge!

_____
**Use <code>for</code>, .split(), and <code>if</code> to create a Statement that will print out words that start with 's':**

In [ ]:
st = 'Print only the words that start with s in this sentence'

In [ ]:
#Code here

______
**Use range() to print all the even numbers from 0 to 10.**

In [ ]:
#Code Here

___
**Use a List Comprehension to create a list of all numbers between 1 and 50 that are divisible by 3.**

In [ ]:
#Code in this cell
[]

_____
**Go through the string below and if the length of a word is even print "even!"**

In [ ]:
st = 'Print every word in this sentence that has an even number of letters'

In [ ]:
#Code in this cell

____
**Write a program that prints the integers from 1 to 100. But for multiples of three print "Fizz" instead of the number, and for the multiples of five print "Buzz". For numbers which are multiples of both three and five print "FizzBuzz".**

In [ ]:
#Code in this cell

____
**Use List Comprehension to create a list of the first letters of every word in the string below:**

In [ ]:
st = 'Create a list of the first letters of every word in this string'

In [ ]:
#Code in this cell

### Great Job!

## SYNTAX FOR LOOP

my_iterable = [1,2,3]
for item_name in my_iterable:
   print(item_name)
>> 1
>> 2
>> 3

In [4]:
my_list = [1,2,3,4,5,6,7,8,9,10]
for jelly in my_list:
    print(jelly)

1
2
3
4
5
6
7
8
9
10


In [5]:
my_list = [1,2,3,4,5,6,7,8,9,10]
for jelly in my_list:
    print('HELLO')

HELLO
HELLO
HELLO
HELLO
HELLO
HELLO
HELLO
HELLO
HELLO
HELLO


IN PYTHON % MEANS DIVISION

In [7]:
my_list = [1,2,3,4,5,6,7,8,9,10]
for num in my_list:
    # Check for even number 
    if num % 2==0:
        print(num)

2
4
6
8
10


In [13]:
my_list = [1,2,3,4,5,6,7,8,9,10]
for num in my_list:
    # Check for even number 
    if num % 2==0:
        print(num)
    else:
        print(f'Odd number: {num}')

Odd number: 1
2
Odd number: 3
4
Odd number: 5
6
Odd number: 7
8
Odd number: 9
10


In [5]:
my_list = [1,2,3,4,5,6,7,8,9,10]
list_sum = 0
for num in my_list:
    list_sum = list_sum + num
print(list_sum)

55


## 1ST METHOD
ABOVE IS THE EXAMPLE OF WHITE SPACE WHEN WE ARE NOT MAINTAINING THE INTENDENT
list_sum = 0
for num in my_list:
    list_sum = list_sum + num
print(list_sum)
## OUTPUT IS BELOW
55
## 2ND METHOD
list_sum = 0
for num in my_list:
    list_sum = list_sum + num
    print(list_sum)
## OUTPUT IS BLOW
1
3
6
10
15
21
28
36
45
55
## DIFFERENCE IS OF WHITESPACE PRINT FUNCTION IS IN FOR LOOP
## WHITESPACE / ITENDATION IS SO IMPORTANT IN PYTHON

## FOR LOOP AFTER FOR WE CAN USE VARIABLE OR VALUE ITSELF BOTH R SAME

In [20]:
my_string = 'Hello World'
for letter in my_string:
    print(letter)

H
e
l
l
o
 
W
o
r
l
d


In [21]:
for letter in 'Hello World':
    print(letter)

H
e
l
l
o
 
W
o
r
l
d


In [7]:
n1 = [1,2,3,4,5,6,7,8,9,10]
n2 = [30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45]
for n in n1:
    if n % 2==0 and n % 3!=0:
        print(n)
for m in n2:
    if m % 2==0  and m % 3!=0:
        print(m)

2
4
8
10
32
34
38
40
44


# COMMENT
TECNICALLY WE DONOT NEED TO ACTUALLY ASSIGN A VARIBALE NAME TO THE STRING
YOU COULD DO IT JUST STRAIGHT UP PUT THE STRING HERE
EXAMPLE
my_string = 'Hello World'
REPLACE BY 
(for letter in 'Hello World')
after for varibale (can by any name you like or easier)


In [3]:
for ghghg in 'Hello World':
    print(ghghg)

H
e
l
l
o
 
W
o
r
l
d


AS LONG AS IT'S THE SAME THING INSIDE OF THIS AND YOU RUN THIS YOU'LL GET BACK THE SAME RESULT

AND THE OTHER THING I WOULD LIKE TO STREE IS THAT YOU TECHNICALLY YOU DONOT EVEN NEED TO USE THIS VARIBALE.

In [8]:
for ghghg in 'Hello World':
    print('Cool!')

Cool!
Cool!
Cool!
Cool!
Cool!
Cool!
Cool!
Cool!
Cool!
Cool!
Cool!


## IMP NOTES
IT WILL PRINT "COOL! 11 TIME AS THE TEXT LENGHT IS 11

## FOR TUPLES PATTERN IS SAME NO DIFFERENCE

In [11]:
tup = (1,2,3)
for item in tup:
    print(item)

1
2
3


## ABOUT TUPLE
Now i want to metnion tuple packing so tupe have a litte bit of a special quality when it comes to for loops
So iterating through a sequence that contains itself self tuples the item can be used with tuple and packing.

In [12]:
mylist = [(1,2),(3,4),(5,6),(7,8)]

In [13]:
len(mylist)

4

In [14]:
for item in mylist:
    print(item)

(1, 2)
(3, 4)
(5, 6)
(7, 8)


This sort of data structure are extremly common in Python and later on we're to see that a lot of build in function
actually take advantage of this built in structure and return back tuples inside a list and reason for that is somthing
called tuple and packing

In [15]:
for (a,b) in mylist:
    print(a)
    print(b)

1
2
3
4
5
6
7
8


## ABOVE EXAMPLE
it will prnt 1 - 8
well this is called the tuple unpacking where you kind of a duplicate the structure of the item.

now for example if i print only a what happen below.

In [16]:
for (a,b) in mylist:
    print(a)

1
3
5
7


In [17]:
for (a,b) in mylist:
    print(b)

2
4
6
8


In [21]:
mylist = [(1,2,3),(5,6,7),(8,9,10)]
for item in mylist:
    print(item)

(1, 2, 3)
(5, 6, 7)
(8, 9, 10)


In [23]:
for a,b,c in mylist:
    print(b)

2
6
9


## NOTE IN ABOVE EXAMPLE
print(a) or print(b) or print(c)
1,5,8       2,6,9       3,7,10

## DICTONARY


In [25]:
d = {'k1':1,'k2':2,'k3':3}
for item in d:
    print(item)

k1
k2
k3


## ABOVE ITEM IS OF KEY AND FOR KEY CONTENT USE THE BELOW SYNTAX
PLEASE USE ITEMS

In [27]:
d = {'k1':1,'k2':2,'k3':3}
for item in d.items():
    print(item)

('k1', 1)
('k2', 2)
('k3', 3)


WHAT YOU NEED TO DO IS CALL THE ITEMS HERE AND NOW WHEN YOU RUN THAT YOU GET BACK THERE TWO PAIRS K-1

In [29]:
d = {'k1':1,'k2':2,'k3':3}
for key,value in d.items():
    print(value)

1
2
3


## ABOVE ITEM IS OF KEY AND FOR KEY CONTENT USE THE BELOW SYNTAX
#### PLEASE USE ITEMS
#### DICTIONARY ARE UNORDERED
#### BUT IF YOU HAVE A VERY LARGE DICTIONARY THERE'S NO GUARANTEE THAT THINGS ARE GOING TO BE SORTED OR IN SORT OF ORDER 

## WHILE LOOPS
WHILE LOOPS WILL CONTINUE TO EXECUTE BLOCK OF CODES WHILE SOME CONDITION REMAINS TRUE.

FOR EXAMPLE, WHILE MY POOL IS NOT FULL, KEEP FILLING MY POOL WITH WATER.

OR MY DOGS ARE STILL HUNGRY, KEEP FEEDING MY DOGS.

# SYNTAX WHILE LOOPS
### while some_boolean_conditions:
       #### do something
   
# you can combine with an else if you want

##### while some_boolean_conditions:
       #### do something
##### else:
       #### do something different

In [4]:
x = 0
while x < 5:
    print(f'The current value of x is {x}')
    # x = x + 1 (this is first pattern).
    x+=1
    #  && second pattern is above.

The current value of x is 0
The current value of x is 1
The current value of x is 2
The current value of x is 3
The current value of x is 4


In [9]:
# USING ELSE STATEMENT AS UNDER
x = 0
while x < 5:
    print(f'The current value of x is {x}')
    # x = x + 1 (this is first pattern).
    #  && second pattern is above.
    x+=1
else:
    print("X IS NOT LESS THEN 5")

The current value of x is 0
The current value of x is 1
The current value of x is 2
The current value of x is 3
The current value of x is 4
X IS NOT LESS THEN 5


In [10]:
# USING ELSE STATEMENT AS UNDER
x = 50
while x < 5:
    print(f'The current value of x is {x}')
    # x = x + 1 (this is first pattern).
    #  && second pattern is above.
    x+=1
else:
    print("X IS NOT LESS THEN 5")

X IS NOT LESS THEN 5


# BREAK, CONTINUE, PASS
##### WE CAN USE BREAK, CONTINUE, PASS STATEMENT
WE CAN USE BREAK, CONTINUE, PASS STATEMENT IN OUR LOOP TO ADD ADDITIONAL FUNCTIONALIY FOR VARIOUS CASES. 
THE 3 STATEMENT DEFINE AS FOLLOW.

break: Break out of the current closest enclosing loop.
continue : Goes on the top of the closest loop.
pass : Does nothing to all.

In [1]:
x = [1,2,3]
for item in x:
    # comment

SyntaxError: unexpected EOF while parsing (<ipython-input-1-e1d473db94f4>, line 3)

ABOVE WILL SHOW ERROR REASON FOR LOOP NOT COMPLETE
# :COMPLIER IS EXPECTING SOME THINGS

In [2]:
x = [1,2,3]
for item in x:
    # comment
    pass
print('end of the script')

end of the script


In [10]:
mystring = 'Sammy'
for letter in mystring:
    if letter == 'a':
        continue
    print(letter)

S
m
m
y


##### IF THE LETTER IS EQUAL TO LOWERCASE A COLON CONTINUE 

In [11]:
mystring = 'Sammy'
for letter in mystring:
    if letter == 'a':
        break
    print(letter)

S


#### LOOP GET BREAK WHEN THE CONDITION MEET THE CRITIERA

In [14]:
x = 0

while x <5 :
    if x==2:
        break
    print(x)
    x += 1

0
1


# INTERUPT CTRL+C

CHAPTER FILES.

Before we finish this section, let's quickly go over how to perform Simpe I/O basic.txt files.

We'll will also discuss file paths on your computer.

Let get started!
    PLEASE NOTE BELOW CODE WORK ONLY IN THE JUPITER NOTEBOOK OR SUBLIME EDITOR.

In [1]:
%%writefile myfile.txt
Hello this is the text file
This is the second line
This is the third line

Writing myfile.txt


In [2]:
myfile = open('myfile.txt')

PLEASE NOTE IF FILE NAME IS WRONG WHAT THE ERROR IS DISPLAY.

In [3]:
myfile = open('whoops_wrong.txt')

FileNotFoundError: [Errno 2] No such file or directory: 'whoops_wrong.txt'

    HOW TO GET THE PATH BELOW IS FUNCTION

In [4]:
pwd

'C:\\Users\\Lenovo'

In [5]:
myfile = open('myfile.txt')

In [ ]:
HOW TO READ THE FILE IS CREATED BELOW IS THE FUNCTION.

In [6]:
myfile.read()

'Hello this is the text file\nThis is the second line\nThis is the third line\n'

In [7]:
myfile.read()

''

SAME VARIBEL IS USE FOR READ SEE WHAT IS DISPLAY.
SO LOT OF TIME BEGINEERS ARE SCRACTING THEIR HEADS THINKING WELL WHAT THE HECK HAPPENDED.
I JUST READ IT NOW WHEN I TRY TO READ AGAIN IT'S NO LONGER THERE.
THE REASON THIS IS HAPPENING IS BECAUSE YOU CAN IMAGINE THAT THERE'S IS A CURSOR AT THE BEGINING OF THE FILE AND WHEN YOU READ IT THE CURSOR GOES AWAY TO THE END OF THE FILE AND YOU NEED TO RESET THE CURSOR OR SEEK BACK TO ZERO IN ORDER TO READ IT AGAIN.
FOR THE YOU NEED TO SAY
myfile = seek(0)

PLEASE NOTE EVERY TIME YOU HAVE SET THE CURSOR TO 0 ELSE IT WILL DISPLAY ''

In [9]:
myfile.seek(0)

0

In [10]:
myfile.read()

'Hello this is the text file\nThis is the second line\nThis is the third line\n'

In [14]:
myfile.seek(0)
contents = myfile.read()

In [15]:
contents

'Hello this is the text file\nThis is the second line\nThis is the third line\n'

In [16]:
myfile.seek(0)

0

    THIS IS ANOTHER FUNCTION READLINE AND SEE THE BELOW OUTPUT.

In [17]:
myfile.readlines()

['Hello this is the text file\n',
 'This is the second line\n',
 'This is the third line\n']

NOW THE CHAPTER IS REGARDING FILE LOCATION

IF WANT TO OPEN FILE AT ANOTHER LOCATION ON YOUR COMPUTER, SIMPLY PASS IN THE ENTIRE FILE PATH

FOR WINDOWS YOU NEED TO USE THE DOUBLE \ SO THE PYTHON DOESN'T TREAT THE SECOND \ AS AN ESCAPE CHARACTER, A FILE PATH IN THIS FORM

myfile = open("C:\\Users\\Lenovo\\myfile.txt")


In [18]:
myfile = open("C:\\Users\\Lenovo\\myfile.txt")

In [22]:
myfile.seek(0)

0

In [23]:
myfile.readlines()

['Hello this is the text file\n',
 'This is the second line\n',
 'This is the third line\n']

WHEN FILE IS OPEN AND WE HAVE TO CLOSE BELOW IS THE SYNTAX

In [24]:
myfile.close()

WHILE PROGRAM IS RUNNING AND SOMEONE DELETE THE FILE, TO TAKE PRECAUTION USING THE WITH STMT
WITH STMT BELOW MENTION.

SO THIS IS THE OLD WAY OF DOING THINGS AND I BELIEVE IT WAS CALLED MY FILE THAT DXi.
WHILE USING WITH STMT
1 AND NOTICE HAPPENS IF I HIT ENTER.

I HAVE AN INDENTATION AND BASICALLY WHAT IT MEANS IS THE ANY CODE HERE THAT'S WITHIN THIS BLOCK THIS IDENTED BLOCK OF CODE IS GOING TO USE MY NEW FILE AS THE VARIABLE OF THIS FILE SO THAT I CAN SAY


In [25]:
with open('myfile.txt') as my_new_file:
    contents = my_new_file.read()
    

In [26]:
contents

'Hello this is the text file\nThis is the second line\nThis is the third line\n'

TO GET THE FUNCTION OF OPEN PRESS SHIFT + TAB AND YOU WILL GET THE GREAT HELP OF OPEN

In [1]:
with open('myfile.txt',mode='r') as myfile:
    contents = myfile.read()

In [2]:
contents

'Hello this is the text file\nThis is the second line\nThis is the third line\n'

In [3]:
with open('myfile.txt',mode='w') as myfile:
    contents = myfile.read()

UnsupportedOperation: not readable

    Reading Writing,Appending Modes
*  mode = 'r'  is read only.
*  mode = 'w'  is write only(will overwrite files or create new!).
*  mode = 'r+' is reading and writing.
*  mode = 'w+' is writing and reading (Overwriting existing files or create new file!).

In [4]:
%%writefile myfile.txt
Hello this is the text file
This is the second line
This is the third line

Overwriting myfile.txt


    HOW TO USING PRINT COMMAND TO PRINT THE TEXT FILE

In [5]:
with open('myfile.txt',mode='r') as f:
    print(f.read())

Hello this is the text file
This is the second line
This is the third line



    PLEASE NOTE : HOW TO APPEND THE TEXT FILE BELOW IS THE CODE AND CURSOR IS AT THE END

In [6]:
with open('myfile.txt',mode='a') as f:
    f.write('FOUR ON FOURTH')

In [7]:
with open('myfile.txt',mode='r') as f:
    print(f.read())

Hello this is the text file
This is the second line
This is the third line
FOUR ON FOURTH


    NOTE THE MODE = 'W' FILE DOES NOT EXIST THEN IT WILL CREATE FILE BELOW IS THE EXAMPLE AND CAN BE READ AND USING PRINT YOU CAN DISPLAY THE FILE CONTENTS.

In [2]:
with open('asfdsdt.txt',mode='w') as f:
    f.write('I CREATED THIS FILE!')

In [3]:
with open('asfdsdt.txt',mode='r') as f:
    print(f.read())

I CREATED THIS FILE!


In [24]:
with open('test.txt',mode='w') as f:
    f.write('Hello World')
    f.close()

In [25]:
with open('test.txt',mode='r') as f:
    print(f.read())

Hello World


# *ARG AND **KWARGS

In [3]:
def myfunc(a,b):
    # Return 5% of the sum a and b
    return sum ((a,b)) * 0.05
myfunc(40,60)

5.0

In [5]:
def myfunc(a,b,c=0,d=0,e=0):
    # Return 5% of the sum a and b
    return sum ((a,b,c,d,e)) * 0.05
myfunc(40,60,100,100,3)

15.15

In [6]:
def myfunc(*args):
    return sum(args) * 0.05

In [8]:
myfunc(40,60,100,1,34)

11.75

# (*) MEANS ANY NAME MULTIPLE ARGUMENTS & PRINT THEM 

In [11]:
def myfunc(*args):
    print(args)

In [12]:
myfunc(40,60,100,1,34)

(40, 60, 100, 1, 34)


In [13]:
def myfunc(*spam):
    print(spam)

In [14]:
myfunc(40,60,100,1,34)

(40, 60, 100, 1, 34)


In [15]:
def myfunc(*args):
    for item in args:
        print(item)

In [16]:
myfunc(40,60,100,1,34)

40
60
100
1
34


In [21]:
def myfunc(**kwargs):
    if 'fruit' in kwargs:
        print('My fruit choice is {}'.format(kwargs['fruit']))
    else:
        print('I did not find any fruit here')

In [22]:
myfunc(fruit='apple')

My fruit choice is apple


In [23]:
myfunc(fruit='apple',veggie='lettuce')

My fruit choice is apple


In [25]:
def myfunc(**kwargs):
    print(kwargs)
    if 'fruit' in kwargs:
        print('My fruit choice is {}'.format(kwargs['fruit']))
    else:
        print('I did not find any fruit here')

In [26]:
myfunc(fruit='apple',veggie='lettuce')

{'fruit': 'apple', 'veggie': 'lettuce'}
My fruit choice is apple


In [2]:
def myfunc(*args,**kwargs):
    #print(args)
    #print(kwargs)
    print('I would like {} {}'.format(args[0],kwargs['food']))

In [3]:
myfunc(10,20,30,fruit='orange',food='eggs')

I would like 10 eggs


In [4]:
myfunc(10,20,30,fruit='orange',food='eggs',animal='dog')

I would like 10 eggs


# Excersie
Functions #8: *args
Define a function called myfunc that takes in an arbitrary number of arguments, and returns the sum of those arguments.

Remember, don't run the function, simply provide the definition.

For example, a function that returns a count of the number of arbitrary arguments might look like:

    def myfunc(*args):
        return len(args)

This is all you need to enter!

To give an idea what the above function would look like when tested:

    myfunc(5,6,7,8)
    # Output: 4
Added note: this exercise requires that the function return the sum. Print statements will not work here.

In [5]:
def myfunc(*args):
    return sum(args)

In [6]:
myfunc(2,3,4,5)

14

# Excersie
Functions #9: pick evens

Define a function called myfunc that takes in an arbitrary number of arguments, and returns a list containing only those arguments that are even.

Remember, don't run the function, simply provide the definition.

To give an idea what the function would look like when tested:

    myfunc(5,6,7,8)
    # Output: [6, 8]

Added note: this exercise requires that the function return a list. Print statements will not work here.

In [14]:
def myfunc(*args):
    args = [num2 for num2 in args if num2 % 2 == 0]
    return args

In [15]:
myfunc(2,3,4,5)

[2, 4]

# Exercise
Functions #10: skyline

Define a function called myfunc that takes in a string, and returns a matching string where every even letter is uppercase, and every odd letter is lowercase. Assume that the incoming string only contains letters, and don't worry about numbers, spaces or punctuation. The output string can start with either an uppercase or lowercase letter, so long as letters alternate throughout the string.

Remember, don't run the function, simply provide the definition.

To give an idea what the function would look like when tested:

    myfunc('Anthropomorphism')
    # Output: 'aNtHrOpOmOrPhIsM'

Added note: this exercise requires that the function return a string. Print statements will not work here.

# CODE BELOW
x = "Anthropomorphism"
result_string = ""
index = 0;
for c in x:
    if(index%2 == 0):
        result_string += c.lower()
    else:
        result_string += c.upper()
    index+=1
print(result_string)

In [177]:
def myfunc(mystrg):
    result = ""
    index = 0;
    for c in mystrg:
        if(index%2 != 0):
            result = result + c.upper()
        else:
            result = result + c.lower()
        index = index + 1
        mystrg = result
    return mystrg

In [178]:
myfunc('Anthropomorphism')

'aNtHrOpOmOrPhIsM'

# FUNCTION PRACTICE PROBLEM
# DEEP LEARNING
* LEARNING FUNCTIONS INCREASES YOUR PYTHON SKILLS EXPONENTIALLY.
* THIS ALSO MEANS THAT THE DIFFICULTIES OF PROBLEMS YOU CAN SOLVE ALSO INCREASE DRASTICALLY
* LET'S GET SOME PRACTICE WITH CONVERTING PROBLEM STATEMENT INTO PYTHON CODE.
* WE WILL GO THROUGH A SERIES OF FUNCTION PRACTIVE EXCERCISE
* AFTER THIS LECTURE WE WILL GO THROUGH THE SOLUTION

* THERE IS TWO OPTION FOR THIS MATERIAL:
  * TRY OUT THE EXCERCISE YOURSELF, THEN GO THROUGH THE SOLUTION
  * TREAT THE SOLUTIONS AS A CODE - ALONG LECTURE FOR MORE GUIDANCE AND PRACTICE.

# EXAMPLE
LESSER OF TWO EVENT : We a function that return a lesser of two given numbers, If both numbers are even, but return the greater, if one or both numbers are odd.
lesser_of_two_even(2,4)--> 2
lesser_of_two_even(2,5)--> 5


In [2]:
def lesser_of_two_even(a,b):
    if a%2 == 0 and b%2 == 0:
        if a < b:
            print("EVEN",a)
            result = a
        else :
            print("ODD",b)
            result = b
    else:
        if a > b:
            result = a
        else:
            result = b
    return result

In [103]:
lesser_of_two_even(2,4)

4

In [3]:
lesser_of_two_even(2,5)

5

# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [4]:
def lesser_of_two_even(a,b):
    if a%2 == 0 and b%2 == 0:
        result = min(a,b)
    else:
        result = max(a,b)
    return result

In [5]:
lesser_of_two_even(2,4)

2

In [6]:
lesser_of_two_even(2,5)

5

# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [7]:
def lesser_of_two_even(a,b):
    if a%2 == 0 and b%2 == 0:
        return min(a,b)
    else:
        return max(a,b)

In [8]:
lesser_of_two_even(2,4)

2

In [9]:
lesser_of_two_even(2,5)

5

# NEXT EXAMPLE
* ANIMAL CRACKERS: Write a function take a two-word string and returns True if both words begin with same letter

* animal_creckers('Levelheaded Llama') --> True
* animal_creckers('Crazy Kangaroo') --> False

In [50]:
def animal_crackers(text):
    wordlist = text.split()
    print(wordlist)
    first = wordlist[0]
    second = wordlist[1]
    return first[0] == second[0]

In [51]:
animal_crackers('Levelheaded Llama')

['Levelheaded', 'Llama']


True

In [52]:
animal_crackers('Crazy Kangaroo')

['Crazy', 'Kangaroo']


False

# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [59]:
def animal_crackers(text):
    wordlist = text.split()
    print(wordlist)
    return wordlist[0][0] == wordlist[1][0]

In [60]:
animal_crackers('Levelheaded Llama')

['Levelheaded', 'Llama']


True

In [61]:
animal_crackers('Crazy Kangaroo')

['Crazy', 'Kangaroo']


False

# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW (USING LOWER() AND UPPER() FUNCTIONS

In [65]:
def animal_crackers(text):
    wordlist = text.lower().split()
    print(wordlist)
    return wordlist[0][0] == wordlist[1][0]

In [66]:
animal_crackers('Levelheaded Llama')

['levelheaded', 'llama']


True

In [67]:
animal_crackers('Crazy cat')

['crazy', 'cat']


True

# NEXT EXAMPLE
* MAKE TWENTY: Given two integer return True if the sum of the integer is 20 or if one of the integers is 20. If not return false
* make_twenty(20,10) --> True
* make_twenty(12,8) --> True
* make_twenty(2,3) --> False

In [76]:
def make_twenty(a,b):
    if a+b == 20 or a ==20 or b == 20:
        print(a,b)
    else:
        print(a,b)
    return a,b

In [78]:
def make_twenty(a,b):
    return a+b == 20 or a ==20 or b == 20

In [81]:
make_twenty(2,3)

False

# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [83]:
def make_twenty(a,b):
    if a+b==20:
        return True
    elif a ==20:
        return True
    elif b == 20:
        return True
    else:
        return False

In [84]:
make_twenty(20,10)

True

In [85]:
make_twenty(12,8)

True

In [86]:
make_twenty(2,3)

False

# NEXT LECTURE PROBLEM TO SOLVE

* OLD MACDONALD: Write a function that capitalize the first and fourth letters of a name.

* old_macdonald('macdonald') ==> MacDonald

* Note : 'macdonald'.capitalize() returns 'Macdonald'

In [9]:
def old_macdonald(txt):
    strs =""
    indx = 1
    for a in txt:
        if (indx == 1) or (indx ==4):
            strs = strs + a.upper()
        else:
            strs = strs + a.lower()
        indx = indx + 1
        txt = strs
    print(strs)
    return

In [10]:
old_macdonald('macdonald')

MacDonald


# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [15]:
def old_macdonald(name):
    first_letter = name[0]
    inbetween = name[1:3]
    fourth_letter = name[3]
    rest = name[4:]
    return first_letter.upper() + inbetween + fourth_letter.upper() + rest

In [16]:
old_macdonald('macdonald')

'MacDonald'

In [19]:
def old_macdonald(name):
    first_half = name[:3]
    second_half = name[3:]
    return first_half.capitalize() + second_half.capitalize()

In [20]:
old_macdonald('macdonald')

'MacDonald'

# NEXT LECTURE LEVEL-1 PROBLEM TO SOLVE

* MASTER YODA:Given a sentence, return a sentence with the word reversed

* MASTER_YODA('I am home') --> 'home am I'

* MASTER_YODA('We are ready') --> 'ready are We'

In [92]:
def master_yoda(txt1):
    strg = ""
    strg = txt1[5:] + txt1[1:5] + txt1[0:1]
    print(strg)
    return

In [93]:
master_yoda("I am home")

home am I


In [94]:
master_yoda("We are ready")

e readye arW


In [103]:
def master_yoda(txt1):
    wordlist = txt1.split()
    reverse_word_list = wordlist[::-1]
    return ' '.join(reverse_word_list)

In [104]:
master_yoda("I am home")

'home am I'

In [105]:
master_yoda("We are ready")

'ready are We'

# NEW FUNCTION JOIN() HOW IT WORKS

In [102]:
mylist = ['a','b','c']
print('--'.join(mylist))
print('  '.join(mylist))
print('000000'.join(mylist))

a--b--c
a  b  c
a000000b000000c


# NEXT LECTURE PROBLEM TO SOLVE

* ALMOST THERE: Given an integer n, return True if n is within 10 of either 100 or 200
* almost_there(90)--> True
* almost_there(104)--> True
* almost_there(150)--> False
* almost_there(209)--> True

In [219]:
def almost_there(n):
        return (abs(100-n) <=10) or (abs(200-n) <=10)

In [224]:
almost_there(209)

True

# NEXT LECTURE LEVEL-2 PROBLEM TO SOLVE

# LEVEL 2 PROBLEMS

FIND 33
Given a list of ints, return True if the array contains 3 next to a 3 somewhere
has_33([1,3,3]) --> True
has_33([1,3,1,3]) --> False
has_33([3,1,3]) --> False

In [274]:
def has_33(*ints):
    a = 0
    b = 0
    x = 0
    for x in ints:
        a = ints[x]
        print("increment",ints[x],x)
#        x = x + 1
        print("after increment",ints[x],x)        
        b = ints[x]
        print(ints)
        print(a,b)
        return a == 3 and b ==3

In [277]:
has_33(1,3,1,3)

increment 3 1
after increment 3 1
(1, 3, 1, 3)
3 3


True

In [288]:
def has_33(nums):
    for i in range(0,len(nums)-1):
        if nums[i]==3 and nums[i+1]==3:
            return True
    return False

In [291]:
has_33([3,1,3])

False

# LEVEL 2 PROBLEMS
PAPER DOLL: Given a String, return a string where for every character int he original there are three charcters
paper_doll('Hello') --> 'HHHeeellllllooo'
paper_doll('Mississippi') --> 'MMMiiissssssiiippppppiii'

In [7]:
def paper_doll(text):
    result = ""
    for char in text:
        result += char*3
    return result

In [8]:
paper_doll('Hello')

'HHHeeellllllooo'

In [9]:
paper_doll('Mississippi')

'MMMiiissssssiiissssssiiippppppiii'

# LEVEL 2 PROBLEMS
BLACK JACK: Given three integers between 1 and 11, if their sum is less then or equal to 21, return their sum, if their sum exceeds 21 and there's an eleven, reduce the total sum by 10. Finally, if the sum(even after ajustment) excees 21, return 'BUST'
* blackjack(5,6,7) --> 18
* blackjack(9,9,9) --> 'BUST'
* blackjack(9,9,11) --> 19

In [1]:
def blackjack(a,b,c):
    d = a+b+c
    if a>0 and a <=11 and b>0 and b<=11 and c>0 and c<=11:
        if d <= 21 and (a!=11 and b!=11 and c!=11):
            return d
        else :
            #        elif (a+b+c) >=21 or (a=11 and b=11 and c=11):
            print("SUM EXCEED TO 21",d)
            d = d - 10
            print("SUM REDUCED BY 10",d)
    else:
        print("NO ENTER BETWEEN 1-11",a,b,c)
    return

In [5]:
blackjack(9,9,11)

SUM EXCEED TO 21 29
SUM REDUCED BY 10 19


# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [10]:
def blackjack(a,b,c):
    if sum([a,b,c]) <= 21:
        return sum([a,b,c])
    elif 11 in [a,b,c] and sum([a,b,c]) <=31:
        return sum([a,b,c])-10
    else:
        return 'BUST'

In [12]:
blackjack(5,6,7)

18

In [13]:
blackjack(9,9,9)

'BUST'

In [14]:
blackjack(9,9,11)

19

# LEVEL 2 PROBLEM
* SUMMER OF '69: Return the sum of number in the array, except ignore sections of numbers starting with a 6 and extending to the next 9 (every 6 will be follower by at latest one 9). Return 0 for no numbers.
* summer_69([1,3,5]) --> 9
* summer_69([4,5,6,7,8,9]) --> 9
* summer_69([2,1,6,9,11]) --> 14

In [9]:
def summer_69(arr):
    total = 0
    add = True
    for num in arr:
        while add:
            if num!=6:
                total += num
                break
            else:
                add = False
        while not add:
            if num !=9:
                break
            else:
                add = True
                break
    return total

In [10]:
summer_69([1,3,5])

9

In [7]:
summer_69([4,5,6,7,8,9])

9

In [11]:
summer_69([2,1,6,9,11])

14

## LEVEL 2 PROBLEM
* SPY GAME: Write a functin that take in a list of integers and returns True if it containts 007 in order
* spy_game([1,2,4,0,0,7,5]) --> True
* spy_game([1,0,2,4,0,5,7]) --> True
* spy_game([1,7,2,0,4,5,0]) --> False

In [60]:
def spy_game(ara):
    a=0
    b=0
    c=0
    rtrn = True
    for n1 in ara:
        if n1==0 or n1==7:
            rtrn = True
            print(n1)
#        else:
#            rtrn = False
#        if n1==7:
#            rtrn = True
#        else:
#            rtrn = False
    return rtrn

In [61]:
spy_game([1,2,4,0,0,7,5])

0
0
7


True

In [62]:
spy_game([1,0,2,4,0,5,7])

0
0
7


True

In [63]:
spy_game([1,7,2,0,4,5,0])

7
0
0


True

# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [5]:
def spy_game(nums):
    code = [0,0,7,'x']
    # [0,0,7,'x']
    # [7,'x']
    # ['x'] length = 1
    for num in nums:
        if num == code[0]:
            code.pop(0)
    return len(code)==1

In [6]:
spy_game([1,2,4,0,0,7,5])

True

In [7]:
spy_game([1,0,2,4,0,5,7])

True

In [8]:
spy_game([1,7,2,0,4,5,0])

False

### LEVEL 2 PROBLEMS
* COUNT PRIMES: Write a functin that returns the number of prime numbers that exist up to and including a given number
* By convention we'll treat 0 and 1 is not prime.
* count_prime(100) --> 25

In [3]:
def count_primes(num):
    # Check for 0 and 1 input
    if num < 2 :
        return 0
    ###############
    # 2 is greater
    ##############
    
    # Store a prime numbers using list keywords
    primes =[2]
    # count going up to the input num
    x = 3
    # x is going thought the every number up to a input number
    while x <= num:
        # Check x is the prime numbers
        for y in range(3,x,2):
            if x%y==0:
                x +=2
                break
        else:
# We have use else keyword in for loop and LIST is append           
            primes.append(x)
            x+=2
    print(primes)
    return len(primes)

In [4]:
count_primes(100)

[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]


25

# ANOTHER WAY TO SOLVE THE PROBLEM IS BELOW

In [5]:
def count_primes(num):
    # Check for 0 and 1 input
    if num < 2 :
        return 0
    ###############
    # 2 is greater
    ##############
    
    # Store a prime numbers using list keywords
    primes =[2]
    # count going up to the input num
    x = 3
    # x is going thought the every number up to a input number
    while x <= num:
        # Check x is the prime numbers
        for y in primes:
            if x%y==0:
                x +=2
                break
        else:
# We have use else keyword in for loop and LIST is append           
            primes.append(x)
            x+=2
    print(primes)
    return len(primes)

In [6]:
count_primes(100)

[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]


25

# HERE TOPIC SESSION ENDS